File: 백준/Silver/6603. 로또/로또.cc
#include <bits/stdc++.h>

using namespace std;

int n;
int arr[14], ans[14];
void rec(int inx, int lev);

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
	while(1){
		cin >> n;
		if(n==0) break;
		for(int i=0;i<n;i++)
			cin >> arr[i];
		rec(0, 0);
		cout << endl;
	}
	return 0;
}

void rec(int inx, int lev){
	if(lev==6){
		for(int i=0;i<6;i++)
			cout << ans[i] << " ";
		cout << "\n";
		return;
	}
	for(int i=inx;i<n;i++){
		ans[lev] = arr[i];
		rec(i+1, lev+1);
	}
}

File: 백준/Silver/6603. 로또/로또.py
def rec(inx, lev):
	global arr, ans, n
	if lev==6:
		for i in ans:
			print(i, end=" ")
		print()
		return
	for i in range(inx, n):
		ans.append(arr[i])
		rec(i+1, lev+1)
		ans.pop()

while True:
	ans = []
	I = list(map(int, input().split()))
	n = I[0]
	arr = I[1:]
	if n==0:
		break
	rec(0, 0)
	print()